x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -104,7 +104,7 @@ public final class Authentication implements ToXContentObject {
     private static final Logger logger = LogManager.getLogger(Authentication.class);
     private static final TransportVersion VERSION_AUTHENTICATION_TYPE = TransportVersion.fromId(6_07_00_99);
 
-    public static final TransportVersion VERSION_REMOTE_ACCESS_REALM = TransportVersion.V_8_7_0;
+    public static final TransportVersion VERSION_CROSS_CLUSTER_ACCESS_REALM = TransportVersion.V_8_8_0;
```

Comment:
I bumped the version here to deal with the "fake" breaking change, instead of introducing a new constant because it's starting to feel like there are too many version constants that are largely irrelevant to the final release. Any version before the refactor in this PR won't properly handle cross cluster access because it either does not support it, or supports it under a different realm name, so setting it to 8.8 here seems like a better bet.

Reply:
I think it would be better to consolidate the version barriers related to RCS into a single one, i.e. the actual version when the feature is released. We can revisit this close to release time.